test/test_helpers/certificates.rb

```diff
@@ -0,0 +1,67 @@
+# Copyright © 2020 Nicky Peeters
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the “Software”), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions
+# of the Software.
+# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+require 'rubygems'
+require 'openssl'
+
+module TestHelpers
+  module Certificates
+    def create_key(alg)
+      case alg
+      when :rsa
+        OpenSSL::PKey::RSA.new(2048)
+      when :dsa
+        OpenSSL::PKey::DSA.new(2048)
```

Comment:
I believe DSA is generally deprecated. Better don't add support for it.

Reply:
I don't think we can decide what is supported or not, because the given key is simply passed to `OpenSSL::PKey` (https://ruby-doc.org/stdlib-3.0.2/libdoc/openssl/rdoc/OpenSSL/PKey.html) which supports DSA, at least for Ruby 3.0.2

Reply:
I think it is safe to support only non-deprecated keys. We can add if anybody requires it. It is deprecated for a reason and supporting it can only require more cumbersome process to remove later.

Reply:
But how we detect the key type and size? I don't think it's straightforward, it would probably require it's own PR.

My idea was to wait for a newer Ruby version where DSA is not supported, so we stop supporting it in a future Ruby update.

Reply:
ok, this is for testing, I first didn't realize, in production code we don't deal with it